src/context-selection/attribute-option-combo-selector-bar-item/use-selected.js

```diff
@@ -2,14 +2,39 @@ import { useCallback } from 'react'
 import { useAttributeOptionComboSelection } from '../use-context-selection/index.js'
 import omit from './omit.js'
 
-export default function useSelected() {
+export default function useSelected(relevantCategoriesWithOptions) {
     const [attributeOptionComboSelection, setAttributeOptionComboSelection] =
         useAttributeOptionComboSelection()
 
-    const deselectAll = useCallback(
-        () => setAttributeOptionComboSelection(undefined),
-        [setAttributeOptionComboSelection]
-    )
+    const deselectAll = useCallback(() => {
+        const nextAocSelection = Object.fromEntries(
+            Object.entries(attributeOptionComboSelection).filter(
+                ([categoryId, optionId]) => {
+                    return relevantCategoriesWithOptions.find(
+                        (category) =>
+                            category.id === categoryId &&
+                            category.categoryOptions.find(
+                                (categoryOption) =>
+                                    categoryOption.id === optionId
+                            )
+                    )
+                }
+            )
+        )
+
+        if (
+            JSON.stringify(nextAocSelection) ===
+            JSON.stringify(attributeOptionComboSelection)
+        ) {
+            return
+        }
+
+        return setAttributeOptionComboSelection(nextAocSelection)
```

Comment:
I find it a bit confusing to return a void/undefined. I would expect just 
```javascript
setAttributeOptionComboSelection(nextAocSelection)
```.
and no return in that case. I started to look into the return value of `setAttributeOptionComboSelection`.